src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -415,6 +415,12 @@ struct program_node {
         cur_id = 0;
     }
 
+    format::type get_required_input0() const { return required_input0; }
+    format::type get_required_output() const { return required_output; }
```

Comment:
`get_preferred_output_format(size_t idx = 0)`
`get_preferred_input_format(size_t idx = 0)`

Reply:
Agree to generalize the function spec. I think `required` is more appropriate name than `preferred` in this case because we will just use the format.

Reply:
Do you think we'll never change the format selected by individual node? For example, if selected node formats lead to reorders between each node, then we may enforce less optimal format for part of the nodes to avoid reorders.
Anyway, "required" makes me think that the node won't work with another format, but in fact it's just our performance optimization.

Reply:
OK I see your point. Let me revisit this logic.